File: add_base_tag.js
const fs = require('fs');
const path = require('path');

function walkDir(dir) {
  const files = fs.readdirSync(dir);
  
  files.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    
    if (stat.isDirectory()) {
      walkDir(filePath);
    } else if (file.endsWith('.html')) {
      let content = fs.readFileSync(filePath, 'utf-8');
      
      // Check if base tag already exists
      if (!content.includes('<base')) {
        // Add base tag after <head
        content = content.replace(
          /<head[^>]*>/i,
          match => match + '\n    <base href="/public/">'
        );
        
        fs.writeFileSync(filePath, content, 'utf-8');
        console.log(`✓ Updated ${filePath}`);
      }
    }
  });
}

walkDir('public');
console.log('Done! Base tags added to all HTML files.');

File: add_base_tag.py
import os
import re

def add_base_tag(html_content):
    # Check if base tag already exists
    if '<base' in html_content:
        return html_content
    
    # Add base tag after <head
    return re.sub(
        r'(<head[^>]*>)',
        r'\1\n    <base href="/public/">',
        html_content,
        flags=re.IGNORECASE
    )

# Add base tag to all HTML files
for root, dirs, files in os.walk('public'):
    for file in files:
        if file.endswith('.html'):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
            
            modified = add_base_tag(content)
            
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(modified)
            
            print(f"✓ Updated {filepath}")

print("Done! Base tags added to all HTML files.")
